### ISSUE TITLE
Discussion about the Translate API

### ISSUE THREAD

**Issue by NoNameProvided** [driftyco/ionic#7691]
#### Short description of the problem:

Ionic 2 has a translate API which is a nice to have functionality. Sadly it takes the natural language based approach when the default language is used as the key for other translations. I think this is a bad approach, which at least leads hard to maintain language files, and default language "lock in".

A few of the issues with natural language based translations: 

- **Possibility of key conflict.** The same sentence is used in multiple places in different context. Which cant require different handling in different languages. 
- **The "key" changes when you change a letter.**  
   This usually cause sentences will appear in the default language if the developer forgets to change the  key sentence in the translations as well. 
- **You must decide your app default language before you start to build it.** Later it will serious work to rewrite you whole UI to use an another language as default.
- **Dynamic sentences are event supported?** Like: `You have selected {{x}} products.`

 Luckily seems the implementation of Translate API is only at its beginning, and I haven't seem wide adoption yet. So I think this is the best time to have some discussion about the Translate API.

#### What behavior are you expecting?

Using a key based approach whether the language files are `.json` or simple `.ts` files exporting an object of strings.  

This would fix the issues above like ensuring the uniqueness of the key over time.

#### Solution?

Of course there is a lot to talk about if this topic is open for discussion, like name spacing and other goodies, but first thing first, I would like to know the team thoughts on the topic.

**Which Ionic Version?** 1.x or 2.x

Ionic 2

**Comment by jgw96** [comment]
Hello! Thanks for opening an issue with us! You are definitely correct about our translation module, it is very early, has some issues, and is very bare bones. The reason this has not been developed is because we are currently waiting on the official Angular 2 way of doing localization, and did not want to write our own implementation as wed rather use the Angular 2 way of doing things for localization. I am currently looking into our options when it comes to translation in an Ionic 2 app, and hope to have something put together soon. Thanks for using Ionic!

**Comment by jgw96** [comment]
Hello @NoNameProvided ! We have decided to recommend that users use https://github.com/ocombe/ng2-translate for all your localization needs. I am currently working on a blog post and a resources article on how to use it with Ionic 2. Thanks! Ill be closing this issue for now, but i will update it with a link to my blog post and resources article when that is done. Thanks!

**Comment by NoNameProvided** [comment]
Alright. I wish Ionic would have an official one tho. While [ocombe/ng2-translate](https://github.com/ocombe/ng2-translate) does the job, it lacks flexibility in some areas, I hoped the Ionic version will fix that.  

So no other option left then contributing them to ng2-translate. 😄

**Comment by jgw96** [comment]
@NoNameProvided I would love to discuss what areas of it you see a lack of flexibility!

**Comment by ocombe** [comment]
Me too, if you see things that need to be improved, don't hesitate to make suggestions !
The next big change that I will make will be to offer more loaders than just the static loader
